### PR TITLE
[curl] Update version to 8.1.2

### PR DESCRIPTION
**Pull request by dg0yt**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/curl/portfile.cmake [hunk 1]
```diff
@@ -14,7 +14,6 @@ vcpkg_from_github(
         mbedtls-ws2_32.patch
         export-components.patch
         0023-fix-find-cares.patch
-        version-major-7-8.patch
```
Comment (dg0yt) [inline ports/curl/portfile.cmake:17]: Integrated upstream.

Comment (gaoyangxiaozhu) [reply ports/curl/portfile.cmake:17]: can anyone help reviw this PR ,we want to install 8.1.2 curl also via VCPKG

**Comment by JonLiu1993** [comment]
@dg0yt, Thanks for your PR, when I tested the feature by command `./vcpkg install curl[*] --allow-unsupported`, I got this error:
````
[1/2] "C:/Program Files/CMake/bin/cmake.exe" -E chdir "../../x86-windows-dbg" "C:/Program Files/CMake/bin/cmake.exe" "F:/Feature-test/curl/vcpkg/buildtrees/curl/src/curl-8_1_2-48a3152245.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=F:/Feature-test/curl/vcpkg/packages/curl_x86-windows/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DCMAKE_PROJECT_INCLUDE=F:/Feature-test/curl/vcpkg/ports/curl/cmake-project-include.cmake" "-DUSE_NGHTTP2=ON" "-DCURL_USE_WOLFSSL=ON" "-DCURL_USE_OPENSSL=ON" "-DCURL_USE_MBEDTLS=ON" "-DCURL_USE_LIBSSH2=ON" "-DBUILD_CURL_EXE=ON" "-DENABLE_ARES=ON" "-DCURL_WINDOWS_SSPI=ON" "-DCURL_BROTLI=ON" "-DCURL_USE_SCHANNEL=ON" "-DCURL_USE_SECTRANSP=ON" "-DUSE_LIBIDN2=ON" "-DUSE_WIN32_IDN=ON" "-DUSE_WIN32_LDAP=ON" "-DENABLE_WEBSOCKETS=ON" "-DHTTP_ONLY=OFF" "-DCURL_DISABLE_LDAP=OFF" "-DPKG_CONFIG_EXECUTABLE=F:/Feature-test/curl/vcpkg/downloads/tools/msys2/5b9374e6ec2c4e17/mingw32/bin/pkg-config.exe" "-DCURL_CA_PATH=none" "-DCURL_CA_BUNDLE=none" "-DENABLE_UNICODE=ON" "-DBUILD_TESTING=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_CA_FALLBACK=ON" "-DCURL_USE_LIBPSL=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/Feature-test/curl/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/curl/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=F:/Feature-test/curl/vcpkg" "-D_VCPKG_INSTALLED_DIR=F:/Feature-test/curl/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DENABLE_DEBUG=ON"
FAILED: ../../x86-windows-dbg/CMakeCache.txt 
"C:/Program Files/CMake/bin/cmake.exe" -E chdir "../../x86-windows-dbg" "C:/Program Files/CMake/bin/cmake.exe" "F:/Feature-test/curl/vcpkg/buildtrees/curl/src/curl-8_1_2-48a3152245.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=F:/Feature-test/curl/vcpkg/packages/curl_x86-windows/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DCMAKE_PROJECT_INCLUDE=F:/Feature-test/curl/vcpkg/ports/curl/cmake-project-include.cmake" "-DUSE_NGHTTP2=ON" "-DCURL_USE_WOLFSSL=ON" "-DCURL_USE_OPENSSL=ON" "-DCURL_USE_MBEDTLS=ON" "-DCURL_USE_LIBSSH2=ON" "-DBUILD_CURL_EXE=ON" "-DENABLE_ARES=ON" "-DCURL_WINDOWS_SSPI=ON" "-DCURL_BROTLI=ON" "-DCURL_USE_SCHANNEL=ON" "-DCURL_USE_SECTRANSP=ON" "-DUSE_LIBIDN2=ON" "-DUSE_WIN32_IDN=ON" "-DUSE_WIN32_LDAP=ON" "-DENABLE_WEBSOCKETS=ON" "-DHTTP_ONLY=OFF" "-DCURL_DISABLE_LDAP=OFF" "-DPKG_CONFIG_EXECUTABLE=F:/Feature-test/curl/vcpkg/downloads/tools/msys2/5b9374e6ec2c4e17/mingw32/bin/pkg-config.exe" "-DCURL_CA_PATH=none" "-DCURL_CA_BUNDLE=none" "-DENABLE_UNICODE=ON" "-DBUILD_TESTING=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_CA_FALLBACK=ON" "-DCURL_USE_LIBPSL=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/Feature-test/curl/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/curl/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=F:/Feature-test/curl/vcpkg" "-D_VCPKG_INSTALLED_DIR=F:/Feature-test/curl/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DENABLE_DEBUG=ON"
-- The C compiler identification is MSVC 19.36.32532.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x86/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: F:/Feature-test/curl/vcpkg/downloads/tools/msys2/5b9374e6ec2c4e17/mingw32/bin/pkg-config.exe (found version "0.29.2") 
-- Checking for module 'libidn2'
--   Found libidn2, version 2.3.4
-- curl version=[8.1.2-DEV]
-- Looking for getch in ws2_32;c-ares::cares
-- Looking for getch in ws2_32;c-ares::cares - found
-- Looking for getch in winmm;ws2_32;c-ares::cares
-- Looking for getch in winmm;ws2_32;c-ares::cares - found
CMake Error at CMakeLists.txt:421 (message):
  CoreFoundation framework not found


-- Configuring incomplete, errors occurred!
````
could you please take a look?

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by dg0yt** [comment]
@JonLiu1993 Please review without `--allow-unsupported`. Unsupported means unsupported.

**Comment by JonLiu1993** [comment]
when I tested the feature by command `./vcpkg install curl[brotli,c-ares,http2,idn,idn2,mbedtls,non-http,openssl,schannel,ssh,sspi,tool,websockets,winidn,winldap,winssl,wolfssl]:x86-windows`, I got this error:
````
FAILED: lib/CMakeFiles/libcurl.dir/vtls/wolfssl.c.obj 
C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1436~1.325\bin\Hostx64\x86\cl.exe   -DBUILDING_LIBCURL -DCURLDEBUG -DCURL_HIDDEN_SYMBOLS -DDEBUGBUILD -DHAVE_CONFIG_H -DLIBSSH2_WIN32 -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -D_UNICODE -D_WINDLL -Dlibcurl_EXPORTS -IF:\Feature-test\curl\vcpkg\buildtrees\curl\src\curl-8_1_2-48a3152245.clean\include -IF:\Feature-test\curl\vcpkg\buildtrees\curl\x86-windows-dbg\lib\..\include -IF:\Feature-test\curl\vcpkg\buildtrees\curl\src\curl-8_1_2-48a3152245.clean\lib\.. -IF:\Feature-test\curl\vcpkg\buildtrees\curl\src\curl-8_1_2-48a3152245.clean\lib\..\include -IF:\Feature-test\curl\vcpkg\buildtrees\curl\x86-windows-dbg\lib\.. -IF:\Feature-test\curl\vcpkg\buildtrees\curl\src\curl-8_1_2-48a3152245.clean\lib -IF:\Feature-test\curl\vcpkg\buildtrees\curl\x86-windows-dbg\lib -external:IF:\Feature-test\curl\vcpkg\installed\x86-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W4 /utf-8 /MP  /MP /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -MDd /showIncludes /Folib\CMakeFiles\libcurl.dir\vtls\wolfssl.c.obj /Fdlib\CMakeFiles\libcurl.dir\ /FS -c F:\Feature-test\curl\vcpkg\buildtrees\curl\src\curl-8_1_2-48a3152245.clean\lib\vtls\wolfssl.c
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\wolfssl/openssl/compat_types.h(54): error C2371: 'EVP_MD': redefinition; different basic types
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\openssl/types.h(106): note: see declaration of 'EVP_MD'
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\wolfssl/openssl/compat_types.h(55): error C2371: 'EVP_MD_CTX': redefinition; different basic types
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\openssl/types.h(107): note: see declaration of 'EVP_MD_CTX'
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\wolfssl/openssl/compat_types.h(56): error C2371: 'EVP_CIPHER': redefinition; different basic types
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\openssl/types.h(104): note: see declaration of 'EVP_CIPHER'
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\wolfssl/openssl/compat_types.h(57): error C2371: 'EVP_CIPHER_CTX': redefinition; different basic types
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\openssl/types.h(105): note: see declaration of 'EVP_CIPHER_CTX'
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\wolfssl/openssl/compat_types.h(58): error C2371: 'ASN1_PCTX': redefinition; different basic types
F:\Feature-test\curl\vcpkg\installed\x86-windows\include\openssl/types.h(82): note: see declaration of 'ASN1_PCTX'
````

**Comment by dg0yt** [comment]
Check if it is a regression. Port curl traditionally supports various ssl backends via features. But I'm not surprised if you can't combine openssl (version 3 in vcpkg) and wolfssl with its openssl compatibility layer (unknown version of openssl).

**Comment by JonLiu1993** [comment]
Feature `brotli,c-ares,http2,idn,idn2,mbedtls,non-http,schannel,sspi,tool,websockets,winidn,winldap,winssl,wolfssl` tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**